### PR TITLE
[Backend] Modified Messages model to inherit board

### LINKED ISSUE #118: - Messages - MessageBoards - Course Inheritance
Implement the inheritance for the msgs, board and courses

### PR DESCRIPTION
**Pull request by shakedhagag**
## Messages - MessageBoards Inheritance

1. Messages model has now a FK of boards
2. MessageBoards model has now a FK of Courses
3. This change of the Model required several changes in related files - 
    -The migration test files that were related to messages.
    -The test files for  each class

closes #118

### CODE REVIEW

File: AcadeMeData/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,5 +1,3 @@
-# Generated by Django 3.2.9 on 2021-12-25 15:57
-
 import django.core.validators
```
Comment (Yarboa) [inline AcadeMeData/migrations/0001_initial.py:1]: Please remove this file from PR. 
It should stay unchanged

File: AcadeMeData/migrations/0003_messageBoards_test_data.py [hunk 2]
```diff
@@ -5,17 +5,25 @@ class Migration(migrations.Migration):
...
         ('AcadeMeData', '0001_initial'),
+        ('AcadeMeData', '0006_university_test_data'),
+        ('AcadeMeData', '0008_Professor_test_data'),
+        ('AcadeMeData', '0009_Course_test_data'),
+
     ]
 
     def generate_data(apps, schema_editor):
-        from AcadeMeData.models import MessageBoards
+        from AcadeMeData.models import MessageBoards, Course, Professor, University
```
Comment (Omeramsc) [inline AcadeMeData/migrations/0003_messageBoards_test_data.py:15]: why is the import inside of the function?

File: AcadeMeData/migrations/0003_messageBoards_test_data.py [hunk 3]
```diff
@@ -5,17 +5,25 @@ class Migration(migrations.Migration):
...
+        university = University(name='Reichman University', location='Herzliya', description='Very nice university')
+        university.save()
+        proffesor = Professor(professor_id=3, name='Shimon Shocken', university=university, description='Nice guy',
```
Comment (Omeramsc) [inline AcadeMeData/migrations/0003_messageBoards_test_data.py:18]: typo in professor

File: AcadeMeData/migrations/0003_messageBoards_test_data.py [hunk 4]
```diff
@@ -5,17 +5,25 @@ class Migration(migrations.Migration):
...
+                              rate=1)
+        course1 = Course(1, 'Open Source Code Workshop', False, 'Build a Django project from the grounds up', proffesor)
```
Comment (Omeramsc) [inline AcadeMeData/migrations/0003_messageBoards_test_data.py:20]: is there a reason we are not saving professor and course?

File: AcadeMeData/migrations/0004_messages_test_data.py [hunk 5]
```diff
@@ -5,17 +5,20 @@ class Migration(migrations.Migration):
     dependencies = [
         ('auth', '__latest__'),
         ('AcadeMeData', '0001_initial'),
+        ('AcadeMeData', '0002_User_test_data'),
+        ('AcadeMeData', '0003_messageBoards_test_data'),
     ]
 
     def generate_msgdata(apps, schema_editor):
-        from AcadeMeData.models import Messages, User
+        from AcadeMeData.models import Messages, User, MessageBoards
```
Comment (Omeramsc) [inline AcadeMeData/migrations/0004_messages_test_data.py:13]: import in function again

File: AcadeMeData/migrations/0004_messages_test_data.py [hunk 6]
```diff
@@ -5,17 +5,20 @@ class Migration(migrations.Migration):
...
         userrnd1 = User.create_user('name', 'email', 'pass', 'kaki', 'cs')
+        board = MessageBoards.get_msgboard_by_id(1)
```
Comment (Omeramsc) [inline AcadeMeData/migrations/0004_messages_test_data.py:15]: not saving user and board?

File: AcadeMeData/migrations/0005_messageTags_test_data.py [hunk 7]
```diff
@@ -11,9 +13,15 @@ class Migration(migrations.Migration):
     ]
 
     def generate_msgtagsdata(apps, schema_editor):
-        from AcadeMeData.models import User, Messages, MessageTags
+        from AcadeMeData.models import User, Messages, MessageTags, University, Professor, Course
```
Comment (Omeramsc) [inline AcadeMeData/migrations/0005_messageTags_test_data.py:16]: import in function

**Review by Yarboa — COMMENTED** [review]
Please update the PR description
no one could learn from it about the what are the changes.
If you can not do it through git. do it through githab edit in the 1st commit description

**Comment by Omeramsc** [comment]
Please ignore my former comments if you saw them, I didn't see the full file and they're invalid.

**Comment by Yarboa** [comment]
Please review the comments and resolve conflicts rebase -> push

File: AcadeMeData/migrations/0001_initial.py (hunk 1 above)
Comment (shakedhagag) [reply AcadeMeData/migrations/0001_initial.py]: done

**Comment by shakedhagag** [comment]
> Please review the comments and resolve conflicts rebase -> push

I rebased and the 2 new tests that were added from the rebase now failing
says that the Client object has no attribute user

**Comment by Yarboa** [comment]
> > Please review the comments and resolve conflicts rebase -> push
> 
> I rebased and the 2 new tests that were added from the rebase now failing says that the Client object has no attribute user

Please refer this :)
https://github.com/redhat-beyond/AcadeMe/pull/102/files
some how it has "gone during rebase"

please contact @DanWaldman to assit

**Review by DanWaldman — APPROVED** [review]
Seems to me you have successfully solved the issue

**Review by DanielItkes — APPROVED** [review]

**Comment by DanielItkes** [comment]
Good work! 
@Yarboa This should fix the other PR's

**Review by Yarboa — APPROVED** [review]
LGTM, nice work

**Pull request merged**